controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -1041,6 +1048,37 @@ func (r *ReconcileHyperConverged) migrateBeforeUpgrade(req *common.HcoRequest) (
 	return upgradePatched, nil
 }
 
+func (r *ReconcileHyperConverged) removeOldApifromCrd(req *common.HcoRequest, crdName, removedApiVersion string) error {
+	var crd apiextensionsv1.CustomResourceDefinition
+	if err := r.client.Get(req.Ctx, types.NamespacedName{Name: crdName}, &crd); err != nil {
+		req.Logger.Error(err, fmt.Sprintf("unable to get %s CRD", crdName))
+		return err
+	}
+
+	if !slices.Contains(crd.Status.StoredVersions, removedApiVersion) {
+		return nil
+	}
+
+	versionIndex := slices.Index(crd.Status.StoredVersions, removedApiVersion)
+	if versionIndex == -1 {
+		return nil
+	}
+
+	patchBytes := []byte(fmt.Sprintf(`[
+					{"op":"test", "path":"/status/storedVersions/%d", "value": "%s"},
+					{"op":"remove", "path":"/status/storedVersions/%d"}]`,
+		versionIndex, removedApiVersion, versionIndex))
```

Comment:
nit: we can use indexes, to not use the same parameters again; e.g.
```suggestion
	patchBytes := []byte(fmt.Sprintf(`[
					{"op":"test", "path":"/status/storedVersions/%[1]d", "value": "%[2]s"},
					{"op":"remove", "path":"/status/storedVersions/%[1]d"}]`,
		versionIndex, removedApiVersion))
```